pkg/controller/migcluster/validation.go

```diff
@@ -193,3 +204,25 @@ func (r ReconcileMigCluster) validateSaSecret(cluster *migapi.MigCluster) (int,
 
 	return 0, nil
 }
+
+// Test the connection.
+func (r ReconcileMigCluster) testConnection(cluster *migapi.MigCluster) (int, error) {
+	if cluster.Spec.IsHostCluster {
+		return 0, nil
+	}
+	_, err := cluster.BuildControllerRuntimeClient(r)
```

Comment:
I don't recall how this function behaves if you give it an invalid set of credentials. I assume it will error out and cause a `TestConnectFailed` status condition to be set?